### ISSUE TITLE
Angles and speeds at turning always zero

### ISSUE THREAD

**Issue by tomvanenckevort** [railnova/osrm-train-profile#7]
_Originally posted by @tomvanenckevort in https://github.com/railnova/osrm-train-profile/issues/6#issuecomment-599099898_

So I have been playing around with this a bit more and decided to have a look at the OSRM tile service to plot the various ways, speeds and angles. 

But when I do that all the speeds and angles displayed at turnings seem to be set to zero:
![image](https://user-images.githubusercontent.com/1218996/76686785-f17bf300-6615-11ea-81dd-b69fddcfb956.png)

I'm not really sure if that is a problem with the OSM data or the way OSRM has processed the data though?

For reference, I'm using the latest `basic.lua` profile from this repo and I run the following commands for OSRM:

- `docker run -t -v "${PWD}:/data" osrm/osrm-backend osrm-extract -p /opt/car.lua /data/cheshire-latest.osm.pbf`
- `docker run -t -v "${PWD}:/data" osrm/osrm-backend osrm-partition /data/cheshire-latest.osrm`
- `docker run -t -v "${PWD}:/data" osrm/osrm-backend osrm-customize /data/cheshire-latest.osrm`

**Comment by C4ptainCrunch** [comment]
Indeed, the turn processing function is very crude for now ([see code](https://github.com/railnova/osrm-train-profile/blob/master/basic.lua#L120-L130)). We only refuse truns that are too sharp and only add a time penalty on u-turns.

![image](https://user-images.githubusercontent.com/85785/77054880-7aa57800-69d0-11ea-8071-bb1f4538ad5e.png)
[Debug map](https://osrm.railnova.eu/debug/#20/50.40751/4.41827)

How would you suggest to improve this ? :slightly_smiling_face:

**Comment by tomvanenckevort** [comment]
Yes, I noticed the angle checks in the profile. While I think that should work fine, when I noticed the angles displayed on the debug map as zero degrees it got me thinking that it would match all of those turns all the time.

So for example:
![image](https://user-images.githubusercontent.com/1218996/77146696-8f8f1380-6a83-11ea-82b0-55ac3867bd5b.png)

The green route here would be fine, but I would want to avoid the red route where it takes a very sharp turn. 
I'm guessing it thinks that turn is zero degrees as well (based on the zero values displayed on the debug map here), so therefore the profile would accept this currently as a valid turn in the route, even though the actual turn is more than 30 degrees.
